USAGOV-2014-Display-Call-Times: Update callCenterWaitTimeLookup.js

IsabelLaurenceau opened the pull request:
<!--- Provide a general summary of your changes in the title above -->
## Jira Task

<!--- Provide a link to the Jira ticket -->
https://cm-jira.usa.gov/browse/USAGOV-2014

## Description
<!--- Summarize the changes made in this pull request, not what it's for. -->
On the phone pages (/phone and /es/llamenos) a new alert should show if the contact center is open and the database has recently been refreshed with data. 

## Type of Changes
<!--- Put an `x` in all the boxes that apply. -->
- [x] New Feature
- [ ] Bugfix
- [ ] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Documentation
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] WWW
  - [ ] Egress
  - [ ] Tools
  - [ ] Cron
- [ ] Other

## Testing Instructions
<!-- This instructions are different from “testing instructions” in Jira – those are typically for Content/UX stakeholders -->
<!-- Not “see Jira” – if they are really the same, copy and paste. -->

### Change Requirements
<!-- Checkboxes to indicate need for changes to some part of the system -->

- [ ] Requires New Documentation (Link: {})
- [ ] Requires New Config
- [ ] Requires New Content

### Validation Steps

- You can verify the page is showing correctly by first going to https://s3-us-gov-west-1.amazonaws.com/cg-4d6fb302-315f-403e-a96e-a7563ccddd3d/1.0/waittime.json
- Note the call times in English and Spanish (these are in seconds) 
- On /phone, /es/llamenos, /chat, and /es/chat go to edit and in the body field change the text format to HTML and add `<div data-call-wait='showTime'>&nbsp;</div>` to the end. 
- If it's -1 the alert box should not be visible on the page (call center is closed) 
- Otherwise: 
-- if it's less than a minute ("en":<60) it should always round to a minute. 
-- if it's more than a minute it should round and show the plural minutes or minutos 

- If the value is not -1 but the alert is not showing verify that the timestamp was taken less than 10 minutes ago. See the checkTimeStamp function I included in the script. 

See examples on stage: 
English -> https://cms-stage.usa.gov/phone
Spanish -> https://cms-stage.usa.gov/es/llamenos


## Security Review
<!-- Checkboxes to indicate need for review -->

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates


## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

IsabelLaurenceau committed:
USAGOV-2014-Display-Call-Times: Update callCenterWaitTimeLookup.js

IsabelLaurenceau committed:
USAGOV-2014-Display-Call-Times: Update callCenterWaitTimeLookup.js

IsabelLaurenceau committed:
Merge branch 'dev' into USAGOV-2014-Display-Call-Times

IsabelLaurenceau committed:
USAGOV-2014-Display-Call-Times: lint

IsabelLaurenceau committed:
USAGOV-2014-Display-Call-Times: Merge branch 'USAGOV-2014-Display-Call-Times' of https://github.com/usagov/usagov-2021 into USAGOV-2014-Display-Call-Times

akf reviewed (changes requested):
I did a cursory review and this looks pretty good. I have one question and a request to make a change: 

- We wanted to display wait times on the chat pages as well -- /chat and /es/chat -- were you planning to do those in a separate PR? (Or were they slipping through the cracks again, which has happened a couple of times before you picked this one up?) 
- The change: you're appending the call wait time to the div with ID `block-usagov-content`. I'd like to have a specific element in the content that needs to be present to have this content added to it, so that someone can go into the CMS and remove it if there's a problem (or move it around if they change the page layout). This was @jacobaaronyeager's idea and I think it was a good one. Thinking about this more, it would make sense for that element to be identified by a data- attribute, something like "data-call-wait='something'". Then you could replace it with a div that had the same data- attribute rather than nesting elements (if nesting is undesirable here). 

A thing I'm doing to test this is, I open web developer tools and set a breakpoint in the callCenterWaitTimeLookup.js script so I can step through it and change the values for number of seconds. (Everything's -1 right now, so that's no fun.)

IsabelLaurenceau committed:
Merge branch 'dev' into USAGOV-2014-Display-Call-Times

IsabelLaurenceau committed:
USAGOV-2014-Display-Call-Times: Get chat time as well

IsabelLaurenceau committed:
USAGOV-2014-Display-Call-Times: Merge branch 'USAGOV-2014-Display-Call-Times' of https://github.com/usagov/usagov-2021 into USAGOV-2014-Display-Call-Times

IsabelLaurenceau committed:
USAGOV-2014-Display-Call-Times: change to replace div

IsabelLaurenceau commented:
Updated.

I previously missed that it also wanted to chat. I have noticed working on this ticket though the chat times are always identical to the phone time. Is this correct? I'm wondering if there is a bug somewhere on the creating the wait time json? 

I also added "On /phone, /es/llamenos, /chat, and /es/chat go to edit and in the body field change the text format to HTML and add `<div data-call-wait='showTime'>&nbsp;</div>` to the end." to the instructions above and on the jira ticket so that the new div can be there

akf reviewed (approved):
Looks good! 

I had also noticed that the chat wait times matched the phone wait times, and I happened to mention to Russell that I thought we might be getting a combined response for both queues. He pointed out that the contact center uses a different system for chat than phone, so we'll need to follow upon whether that's actually valid information. (I'll ask.) 

But we can move forward with this, since we can simply omit the new div from the chat pages if it turns out that's bad information.
